presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AnalyzeFinishNode.java

```diff
@@ -0,0 +1,162 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.plan;
+
+import com.facebook.presto.metadata.AnalyzeTableHandle;
+import com.facebook.presto.metadata.TableHandle;
+import com.facebook.presto.sql.planner.Symbol;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class AnalyzeFinishNode
+        extends PlanNode
+{
+    private final PlanNode source;
+    private final Symbol rowCountSymbol;
+    private final AnalyzeTarget target;
+    private final StatisticAggregationsDescriptor<Symbol> descriptor;
+
+    @JsonCreator
+    public AnalyzeFinishNode(
+            @JsonProperty("id") PlanNodeId id,
+            @JsonProperty("source") PlanNode source,
+            @JsonProperty("target") AnalyzeTarget target,
+            @JsonProperty("rowCountSymbol") Symbol rowCountSymbol,
+            @JsonProperty("descriptor") StatisticAggregationsDescriptor<Symbol> descriptor)
+    {
+        super(id);
+        this.source = requireNonNull(source, "source is null");
+        this.target = requireNonNull(target, "target is null");
+        this.rowCountSymbol = requireNonNull(rowCountSymbol, "rowCountSymbol is null");
+        this.descriptor = requireNonNull(descriptor, "descriptor is null");
+    }
+
+    @JsonProperty
+    public PlanNode getSource()
+    {
+        return source;
+    }
+
+    @JsonProperty
+    public AnalyzeTarget getTarget()
+    {
+        return target;
+    }
+
+    @JsonProperty
+    public StatisticAggregationsDescriptor<Symbol> getDescriptor()
+    {
+        return descriptor;
+    }
+
+    @JsonProperty
+    public Symbol getRowCountSymbol()
+    {
+        return rowCountSymbol;
+    }
+
+    @Override
+    public List<PlanNode> getSources()
+    {
+        return ImmutableList.of(source);
+    }
+
+    @Override
+    public List<Symbol> getOutputSymbols()
+    {
+        return ImmutableList.of(rowCountSymbol);
+    }
+
+    @Override
+    public PlanNode replaceChildren(List<PlanNode> newChildren)
+    {
+        return new AnalyzeFinishNode(
+                getId(),
+                Iterables.getOnlyElement(newChildren),
+                target,
+                rowCountSymbol,
+                descriptor);
+    }
+
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context)
+    {
+        return visitor.visitAnalyzeFinish(this, context);
+    }
```

Comment:
add `toString()`

what will `EXPLAIN ANALYZE ..` output? Do you have a test?

Reply:
Previously I tested it manually. The parser is smart enough to distinguish them. I can add a test.